src/sources/aws_s3/sqs.rs

```diff
@@ -405,8 +405,14 @@ impl IngestorProcess {
     }
 
     async fn handle_sqs_message(&mut self, message: Message) -> Result<(), ProcessingError> {
-        let s3_event: SqsEvent = serde_json::from_str(message.body.unwrap_or_default().as_ref())
-            .context(InvalidSqsMessageSnafu {
+        let mut sqs_body: String = message.body.unwrap_or_default();
+        let sns_notification_res: Result<SnsNotification, _> =
+            serde_json::from_str(sqs_body.as_ref());
+        if let Ok(sns_notification) = sns_notification_res {
+            sqs_body = sns_notification.message;
+        }
+        let s3_event: SqsEvent =
+            serde_json::from_str(sqs_body.as_ref()).context(InvalidSqsMessageSnafu {
```

Comment:
```
let sqs_body = serde_json::from_str::<SnsNotification>(sqs_body.as_ref())
    .map(|notification| notification.message)
    .unwrap_or_else(|| message.body.unwrap_or_default());
```

Something like this seems a little cleaner?

Reply:
🤷 I could go with that, though readability-wise IMO I get tripped up seeing a var named sqs_body being turbofish'd to SnsNotification. That took a few seconds to mentally work out. The mapping chain here makes it read like SQS-SNS-S3 is the normal use-case, and SQS-S3 is the else-path.

Reply:
Maybe a renaming would help? I definitely prefer the mapping to the if/else style personally.

Reply:
Took my aging box a while to catch up. I started making the changes and realized your turbofish'd `from_str` was coming from `sqs_body` not the function's incoming Message param. That didn't quite work, chicken-and-egg. ;)

I've got this now, seem okay before I commit?
```
let sqs_body = message.body.unwrap_or_default();
let sqs_body = serde_json::from_str::<SnsNotification>(sqs_body.as_ref())
    .map(|notification| notification.message)
    .unwrap_or(sqs_body);
```

Reply:
🤔 there might be a way to drop the first line as well, but otherwise - that looks good

Reply:
Yeah I was trying for a while. But you need the same effective value in both the `from_str` parse attempt and the `unwrap_or`. Though the `from_str` would really want `message.body().unwrap_or_default()` (note the parens to get straight to an &str rather than String) vs the `unwrap_or` wanting a String. Got too messy to try and be cute removing the first line.